### PR TITLE
jointplot now respects the ratio argument (top plot is larger) with matplotlib backend

### PR DESCRIPTION
**Pull request by DougBurke**
# Summary

The matplotlib back end now makes the main (top) plot taller than the secondary (bottom) plot when using the sherpa.ui.plot_fit_resid and sherpa.ui.plot_fit_delchi routines (this also holds for the sherpa.astro.ui variants).

# Details

With this change the sherpa[.astro].ui.plot_fit_resid/delchi routines now display with the top plot being twice the height of the bottom plot when using the matplotlib (pylab) backend. The code was just not applying the requested `ratio` argument.

There is no test for this as there is currently no good test suite for the plot contents when a plotting backend is available.

This is to address #587 and collides with #626 (although easy to address).

An example of what `sherpa.astro.ui.plot_fit_delchi` looks like with this PR (examples from before the change are available in #587):

resized

The commits should be squashed into a single commit before merging, but left separated for initial review. There is no documentation on what a "joint plot" is meant to represent, so unclear how to handle the `nrows` and `ncols` arguments to `set_jointplot`.

### CODE REVIEW

**Comment by DougBurke** [comment]
I cancelled the tests for the 4614e4a build because I realized I needed to add a change, and then one of the tests failed due to network timeouts, so it was easier to just cancel the whole build and let the 697430a tests run instead.

**Comment by DougBurke** [comment]
I'm going to rebase this to fix the conflict.

File: docs/fit/index.rst [hunk 1]
```diff
@@ -368,6 +368,30 @@ The residuals plot shows, on the ordinate, :math:`\sigma = (d - m) / e` where
...
 for each bin respectively.
 
+The use of this style of plot - where there's the data and fit
+in the top and a related plot (normally some form of residual about
+the fit) in the bottom - is common enough that Sherpa provides
+a specialization of
+:py:class:`~sherpa.plot.SplitPlot` called
+:py:class:`~sherpa.plot.JointPlot` for this case. In the following
+example the plots from above are re-used, as no settings have
+changed, so there is no need to pcall the
```
Comment (wmclaugh) [inline docs/fit/index.rst]: Do you mean 'call' here?

Comment (DougBurke) [reply docs/fit/index.rst]: I think I probably did.

**Pull request merged**